modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/config/LifecycleAwareConfigurationInstance.java

```diff
@@ -106,6 +115,16 @@ public void initialise() throws InitialisationException
         {
             initStats();
             doInitialise();
+            muleContext.registerListener((CustomNotificationListener<MetadataNotification>) notification -> {
+                try
+                {
+                    muleContext.getRegistry().lookupObject(MuleMetadataManager.class).disposeCache(name);
```

Comment:
This completely defeats the purpose. It's the MetadataMAnager the one which should be listening for a config notification saying that it has been disposed